homeassistant/helpers/script.py

```diff
@@ -86,6 +81,21 @@ def call_from_config(
     Script(hass, cv.SCRIPT_SCHEMA(config)).run(variables, context)
 
 
+async def async_validate_action_config(
+    hass: HomeAssistant, config: ConfigType
+) -> ConfigType:
+    """Validate config."""
+    action_type = _determine_action(config)
+
+    if action_type == ACTION_DEVICE_AUTOMATION:
```

Comment:
Should we validate the other types?

Reply:
Not sure it's needed, other types are still validated through the action schema. We need to validate `ACTION_CONDITION` because it might include device conditions. That's coming in a follow up PR.